backend/controllers/userControllers.js

```diff
@@ -50,4 +50,35 @@ const register = async (request, response) => {
   }
 };
 
-module.exports = { register };
+const login = async (request, response) => {
+  const { usernameOrEmail, password } = request.body;
+
+  if (usernameOrEmail && password) {
+    const querySelectUser =
+      "SELECT * FROM users WHERE (username = $1 OR email = $1)";
```

Comment:
could do SELECT password FROM users instead of grabbing the whole row

Reply:
I could specify fields to be returned, but we can do this (which is already in the file on lines 65 - 70):

request.session.user = {
  id: user.id,
  username: user.username,
  email: user.email,
  profileImage: user.profile_image,
};

When the client with session makes another request, the `request.session.user` should be passed into `response.locals` (in server.js) which can supposedly be referenced in EJS views as just object `user`

Reply:
Nice I see